apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.IntegrationTests/Trace/TraceTest.cs

```diff
@@ -358,26 +357,89 @@ public async Task Trace_SizedBuffer_Stress()
                 await Task.WhenAll(responseTasks);
             }
 
-            // TODO: When #2834 is fixed, we should always get as many traces as requests.
-            // Althouhg in the case of a sized buffer we might be missing the last couple of traces
-            // that din't managed to fill the buffer. But this can be calculated and taken into account exactly.
-            // Note, 40% of traces is unexpectedly low, I'll look into it more once #2834 is fixed.
-            var minTraces = (int)((4.0 / 10) * requests);
+            // We expect only around 85% of traces to have been sent to the backend because of the
+            // size of the buffer, i.e. the last buffer won't flush because it won't reach the
+            // maximum buffer size during this test. That's entirely expected.
+            var minTraces = (int)((8.5 / 10) * requests);
             var traces = _polling.GetTraces(uri, _startTime, minEntries: minTraces);
             Assert.InRange(traces.Count(), minTraces, requests);
         }
+
+        [Fact]
+        public async Task Trace_SizedBufferTinyPropagate_Stress()
+        {
+            var uri = $"/Trace/{_testId}";
+            // Not the best ever stress test but we are limited by read quotas anyway.
+            var requests = 300;
+            IList<Task<HttpResponseMessage>> responseTasks = new List<Task<HttpResponseMessage>>(300);
+            Task<HttpResponseMessage[]> whenAllTask = null;
+            using (var server = new TestServer(new WebHostBuilder().UseStartup<TraceTestTinyBufferHighQpsPropagateApplication>()))
+            using (var client = server.CreateClient())
+            {
+                for (int i = 0; i < requests; i++)
+                {
+                    responseTasks.Add(client.GetAsync(uri));
+                }
+
+                // With such a tiny buffer (it fills with at most two traces), we have a lot of concurrent calls to the
+                // backend trying to send traces. Some of these will timeout, some of these will be succesfull.
+                // By default we don't propagate the exception but we are doing so for the purpose of this test.
```

Comment:
It's not clear to me where we're setting it up to propagate exceptions. Maybe include that in the comment?